core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala

```diff
@@ -316,8 +316,9 @@ trait SubmissionMonitor extends FutureSupport with LazyLogging with RawlsInstrum
         for {
           costBreakdown <- executionServiceCluster.getCost(workflowRec, petUser)
           updatedWorkflowRec <-
-            // TODO CORE-217: if workflow is already in Aborted or Aborting, don't try to re-abort it
-            if (costBreakdown.cost > costCap) {
+            if (
+              costBreakdown.cost > costCap && costBreakdown.status != SubmissionStatuses.Aborted.toString && costBreakdown.status != SubmissionStatuses.Aborting.toString
```

Comment:
This ultimately results in the same string comparison, but it should use `WorkflowStatus`
```suggestion
              costBreakdown.cost > costCap && costBreakdown.status != WorkflowStatuses.Aborted.toString && costBreakdown.status != WorkflowStatuses.Aborting.toString
```